WIP: Daniel and PL's workstream

d4hines opened the pull request:
This is a draft PR so @ulrikstrid can deploy our latest work. Not intended to be merged directly.

Pilou97 committed:
feat: start

And adds some command parsing: nodes, modes and verbosity

Pilou97 committed:
feat: sandbox tear-down

Pilou97 committed:
feat: deposit-deploy dummy-ticket

Pilou97 committed:
feat: sandbox setup

EduardoRFS committed:
helpers: expose Map.S_with_yojson

EduardoRFS committed:
crypto: make BLAKE2B.Map a Map_with_yojson

EduardoRFS committed:
protocol: hold only hash of operation on protocol

EduardoRFS committed:
consensus: produce block before applying

EduardoRFS committed:
protocol: stop checking height on apply_block_data

EduardoRFS committed:
consensus: handle out of order block arrival

Pilou97 committed:
feat: deposit-withdraw-test

Pilou97 committed:
ci: update tilt and nix to use ocaml sandbox

d4hines committed:
feat: load-test

d4hines committed:
feat: liveness check

d4hines committed:
chore: more realistic block and confirmation time

d4hines committed:
refactor: with_validator_uris in deku-node

shouldn't change any behavior

d4hines committed:
feat: allow passing validator uris as option

Deku cluster deployments are very slow while
the nodes discover each other's URI's. This is
needless in the case when you're boostrapping
the chain, as you already know the URI's.

This commit adds a flag that lets you specify the
URI's as a comma-separated list to the node.
If specified, the node will use this list to
broadcast blocks and signatures.

d4hines committed:
wip: big refactor of deku-sandbox

d4hines committed:
feat: allow passing log level via env

d4hines committed:
fix: don't kill nodes after deposit withdraw test

Instead, we should let Tilt do this for us.

d4hines committed:
refactor: files library

d4hines committed:
refactor: sandbox library

d4hines committed:
refactor: extract helpers module

d4hines committed:
refactor: extract a setup module

Pilou97 committed:
fix: remove usage of sleep in withdraw-deposit-test

d4hines committed:
feat: log when applying and producing blocks

d4hines committed:
feat: log block rejection bitmap

Pilou97 committed:
fix: correctly pack and sign discovery payloads

Pilou97 committed:
feat: add command to update discovery contract

d4hines committed:
refactor: extract start module

d4hines committed:
wip: many changes, will clean later

d4hines committed:
wip

aguillon committed:
fix: overengineering the log output

aguillon committed:
fix: killing subprocesses at exit

Pilou97 committed:
refactor: extrat tear-down module

Pilou97 committed:
refactor: extract deposit-withdraw-test module

Pilou97 committed:
refactor: extract deploy-dummy-ticket module

Pilou97 committed:
refactor: extract deposit-dummy-ticket module

Pilou97 committed:
refactor: extract load-test module

Pilou97 committed:
refactor: extract check-liveness module

Pilou97 committed:
refactor: convenient way to add command

Pilou97 committed:
refactor: remove verify function from retry function

d4hines committed:
wip: validator logs

d4hines committed:
wip: update to jakartanet

d4hines committed:
refactor: remove docker mode

It's currently broken and adds a lot of branching

Instead, we should take what we need as
params. Will introduce this in coming commits.

d4hines committed:
feat: accept rpc_address in sandbox commands

d4hines committed:
feat: accept tezos secret in sandbox commands

ulrikstrid commented on sandbox/setup.ml:
```diff
@@ -127,13 +130,14 @@ let setup mode validators rpc_url =
   let consensus = "./src/tezos_interop/consensus.mligo" in
   let discovery = "./src/tezos_interop/discovery.mligo" in
-  let secret = "edsk3QoqBuvdamxouPhin7swCvkQNgq4jP5KZPbwWNnwdZpSpJiEbq" in
+  let secret = "edsk4TxW4UvCXFZrB5ifMx83PAUECKLUB95ecm1Lp4GbE8ZEeE3T1g" in
```
Could we instead default to flextesa and take this as a environment/flag?

ulrikstrid committed:
chore: Update json-reporter for new time format

ulrikstrid committed:
Re-disable non-deku logs

ulrikstrid committed:
Re-add block metrics

d4hines committed:
wip: feat: accept data folder as parameter

d4hines committed:
fix: limit applied blocks to fix memory leak

This feature was always known to be a hack and
a bad memory leak, but now it's affecting us so
it's time to remove it.

d4hines committed:
fixup! refactor: extract a setup module

d4hines committed:
chore: update secret key used in flextesa

It appears the secret key for the "bob" address hard
coded in flextesa has changed.

d4hines committed:
demo: add global counter to core deku state

fraidev committed:
Use 5 seconds for the taquito polling on Production (#760)

* use 5 seconds for the confirmation polling

* use 5 seconds for the streamer polling

d4hines committed:
chore: log rpc address

d4hines committed:
fix: use imported wallet instead of hard-coded

This is necessary because we switched to using the local tezos-client
instead of the one inside flextesa. Hence, the hard-coded flextesa
wallet doesn't exist

d4hines committed:
ci: test
